### PR TITLE
10. get articles by querying

### PR DESCRIPTION
**Pull request by zakaria552**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -210,4 +211,56 @@ describe("PATCH /api/articles/:article_id", () => {
...
+    })
+    test("GET 200 - sorts by given column name", () => {
+        return request(app).get("/api/articles?sort_by=article_id").expect(200).then(({body}) => {
+            expect(body.articles).toBeSortedBy("article_id", {descending: true})
+        })
+    })
+    test("GET 200 - sorts by date in descending order if not given sort by query", () => {
+        return request(app).get("/api/articles").expect(200).then(({body}) => {
+            expect(body.articles).toBeSortedBy("created_at", {descending: true})
+        })
```
Comment (sirKevlar) [inline __tests__/app.test.js:253]: We need one more happy path for if our client specifies an order in the query

File: __tests__/app.test.js [hunk 2]
```diff
@@ -210,4 +211,56 @@ describe("PATCH /api/articles/:article_id", () => {
...
+        return request(app).get("/api/articles?sort_by=password").expect(400).then(({body}) => {
+            expect(body.msg).toBe("invalid sort query!")
+        })
+    })
+    test("GET 400 - given invalid order query returns invalid order query", () => {
+        return request(app).get("/api/articles?sort_by=created_at&order=square").expect(400).then(({body}) => {
+            expect(body.msg).toBe("invalid order query!")
+        })
+    })
+
```
Comment (sirKevlar) [inline __tests__/app.test.js:265]: We need one more sad path test for in any of the keys (sort|_by, order, topic) are misspelled

File: __tests__/app.test.js [hunk 3]
```diff
@@ -210,4 +211,56 @@ describe("PATCH /api/articles/:article_id", () => {
...
     })
+})
+describe("/api/articles(queries)", () => {
+    test("GET 200 - filters the articles by specified topic", () => {
+        return request(app).get("/api/articles?topic=mitch").expect(200).then(({body}) => {
+            expect(body.articles).toHaveLength(11)
+            body.articles.forEach((article) => {
+                expect(article.topic).toBe("mitch")
+            })
+        })
```
Comment (callumu21) [inline __tests__/app.test.js:222]: Just missing one test for a valid topic query that has no associated articles. What should be returned for a topic that exists in the database but has no articles? Currently, your code defaults to an error if the returned array is empty, but this isn't a bad request if the topic query exists in the database.

**Review by callumu21 — CHANGES_REQUESTED** [review]
Nice additions to the test suite. Just missing one more test - see my comments for advice.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -210,4 +211,56 @@ describe("PATCH /api/articles/:article_id", () => {
...
+    })
+    test("GET 200 - given no topic query returns all the articles", () => {
```
Comment (Geo-Beard) [inline __tests__/app.test.js:224]: You still need a test for a valid topic that has no associated articles - i.e. valid topic, status 200, returns an empty array.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -210,4 +211,56 @@ describe("PATCH /api/articles/:article_id", () => {
...
+    test("GET 200 - sorts by given column name", () => {
+        return request(app).get("/api/articles?sort_by=article_id").expect(200).then(({body}) => {
+            expect(body.articles).toBeSortedBy("article_id", {descending: true})
+        })
+    })
+    test("GET 200 - sorts by date in descending order if not given sort by query", () => {
+        return request(app).get("/api/articles").expect(200).then(({body}) => {
+            expect(body.articles).toBeSortedBy("created_at", {descending: true})
+        })
+    })
```
Comment (Geo-Beard) [inline __tests__/app.test.js:254]: Still need a test where /api/articles?order=asc or /api/articles?order=desc. Should be a status 200.

**Review by Geo-Beard — COMMENTED** [review]
Still need 2 more tests - see my comments.

**Pull request closed without merging**